### PR TITLE
Enable unit testing test_matmul_cuda for ROCm

### PR DESCRIPTION
**Pull request by BLOrange-AMD**
test_file | test_name | test_class
-- | -- | --
test_matmul_cuda | test_cublas_addmm_size_10000_cuda_bfloat16 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_10000_cuda_float16 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_10000_cuda_float32 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_1000_cuda_bfloat16 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_1000_cuda_float16 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_1000_cuda_float32 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_100_cuda_bfloat16 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_100_cuda_float16 | (__main__.TestMatmulCudaCUDA)
test_matmul_cuda | test_cublas_addmm_size_100_cuda_float32 | (__main__.TestMatmulCudaCUDA)

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony

### CODE REVIEW

File: test/test_matmul_cuda.py [hunk 1]
```diff
@@ -94,8 +93,8 @@ def test_cublas_addmm(self, size: int, dtype: torch.dtype):
         # Compare
         self.assertEqual(res_cpu, res_cuda)
 
-    @onlyCUDA
```
Comment (pruthvistony) [inline test/test_matmul_cuda.py:97]: any reason to remove this? This UT is only for GPU both CUDA and ROCm.

File: test/test_matmul_cuda.py [hunk 2]
```diff
@@ -39,8 +38,8 @@ def tearDown(self):
         torch.backends.cuda.matmul.allow_tf32 = True
         super(self.__class__, self).tearDown()
 
-    @onlyCUDA
```
Comment (pruthvistony) [inline test/test_matmul_cuda.py:42]: same comment.
any reason to remove this? This UT is only for GPU both CUDA and ROCm.

**Review by pruthvistony — CHANGES_REQUESTED** [review]

**Review by pruthvistony — APPROVED** [review]
LGTM, waiting for CI jobs to finish.

**Comment by jithunnair-amd** [comment]
@malfet Please review

**Review by malfet — COMMENTED** [review]
Please add `ROCM52OrLater` to `torch.testing._internal.common_cuda` and use it throughout the code (as I suspect those would be more frequent in the future)

**Comment by BLOrange-AMD** [comment]
@malfet Thanks for the suggestion. I used skipIfRocmVersionLessThan instead of ROCM52OrLater for easier code maintenance on our side down the road.

**Review by jithunnair-amd — APPROVED** [review]

**Comment by pruthvistony** [comment]
@malfet, 
Can you please help review this PR.

**Review by malfet — APPROVED** [review]

File: test/test_matmul_cuda.py [hunk 3]
```diff
@@ -113,7 +115,8 @@ def test_cublas_addmm_alignment(self):
         self.assertEqual(out, torch.matmul(X, A.transpose(1, 0)) + B)
 
     @onlyCUDA
-    @unittest.skipIf(TEST_WITH_ROCM, "Only CUDA 11+ is supported")
+    @unittest.skipIf((torch.version.cuda is not None) and (not CUDA11OrLater), "Only CUDA 11+ is supported")
+    @skipIfRocmVersionLessThan((5, 2))
```
Comment (malfet) [inline test/test_matmul_cuda.py:119]: Do we really need this check if we are on 5.3/5.4 already?

Comment (BLOrange-AMD) [reply test/test_matmul_cuda.py:119]: We prefer to keep this check, as in our internal CI, we still test upstream PyTorch code with older ROCm versions.

**Review by albanD — CHANGES_REQUESTED** [review]
Something is definitely wrong with the rebase here

**Comment by BLOrange-AMD** [comment]
Closing out this PR. The same fix is now provided in the newly created PR: https://github.com/pytorch/pytorch/pull/98797

**Pull request closed without merging**